### DISCUSSION TITLE
looking for java package com.keyfactor.util;

### DISCUSSION THREAD

**Discussion by syisunny** [Keyfactor/ejbca-ce#484] [Q&A] [answered]
Hello all,
we have an ejb client for ejbca. it seems that from ejbca 8, some util classes are moved out of ejbca project: like CertificateWrapper, CertTools. now they are referenced in com.keyfactor.util. 
could you provide information on how to get this util classes ?
Thank you in advance!
Best regards,
Yi

**Comment by syisunny** [comment]
Found it, these classes are moved to a file called x509-common-util-0.10.9. unfortunately it is a LGPL private internal lib (src and jar are not published), even we could retrieve it from the release src

**Reply by primetomas** [reply, marked as answer]
You can get the jar from EJBCA, it's here: https://github.com/Keyfactor/ejbca-ce/tree/main/lib

I think we may publish the sources a bit later, the componentization is still under construction.
